dora/src/masm/x64.rs

```diff
@@ -417,6 +418,19 @@ impl MacroAssembler {
         }
     }
 
+    pub fn int_shlx(&mut self, mode: MachineMode, dest: Reg, lhs: Reg, rhs: Reg) {
+        let x64 = match mode {
+            MachineMode::Int32 => 0,
+            MachineMode::Int64 => 1,
+            _ => unimplemented!(),
+        };
+
+        emit_vex3_rxbm(self, rhs.msb(), 0, lhs.msb(), VEXM_0F38);
```

Comment:
Implementation of these operations could go into `x64/asm.rs`. But I guess that's a matter of taste. I am trying to refactor the Assembler/MacroAssembler anyways. I am creating an Assembler in `dora/src/asm.rs` and I am going to move all CPU-specific instructions into that directory. So in the future this should be implemented in `dora/src/asm/x64.rs`.

Reply:
> Implementation of these operations could go into x64/asm.rs

Do you mean `emit_vex3_rxbm` etc. or the enclosing function, because the former already are.

Reply:
I was thinking about the enclosing function.

Reply:
Ah, right, thanks!